BendAssist.App/BendAssists/BendDeduction.cs

```diff
@@ -11,20 +13,157 @@ public sealed class BendDeduction : BendAssist {
+      if (mPart is null) return;
+      // Bend Deduction on bend lines
+      newBLines.AddRange (GetTranslatedBLines (bLines, out double totalBD, out int horBLCount, out int verBLCount, isNegOff: blLoc is EBLLocation.Top));
+      // Bend Deduction on edges
+      foreach (var pLine in GetProfilePLines (mPart, blLoc, verBLCount > 0, horBLCount > 0)) {
+         var pLOrient = pLine.Orientation;
+         var (dx, dy) = blLoc is EBLLocation.Top ? pLOrient is Horizontal ? (0.0, -totalBD) : (-totalBD, 0)
+                                                 : pLOrient is Horizontal ? (0.0, totalBD) : (totalBD, 0);
+         newPLines.Add ((PLine)pLine.Translated (dx, dy)); // Parallel pLines
+         switch (blLoc) {
+            case EBLLocation.Top:
+               foreach (var idx in GetCPIndices (pLine, mPart.PLines)) { // Trims the connected perpendicular edges
+                  var conPLine = mPart.PLines.Where (cC => cC.Index == idx).First ();
+                  PLine trimmed;
+                  if (pLOrient is Horizontal)
+                     trimmed = conPLine.StartPoint.Y < pLine.StartPoint.Y ? (PLine)conPLine.Trimmed (0, 0, 0, -totalBD)
+                                                                          : (PLine)conPLine.Trimmed (0, -totalBD, 0, 0);
+                  else
+                     trimmed = conPLine.StartPoint.X < pLine.StartPoint.X ? (PLine)conPLine.Trimmed (0, 0, -totalBD, 0)
+                                                                          : (PLine)conPLine.Trimmed (-totalBD, 0, 0, 0);
+                  tempPLines.Add (trimmed);
+               }
+               break;
+            case EBLLocation.Bottom:
+               foreach (var idx in GetCPIndices (pLine, mPart.PLines)) {
+                  PLine conPLine;
+                  if (tempPLines.Any (tC => tC.Index == idx)) // checks whether the edge is trimmed from the other side or not
+                     conPLine = tempPLines.Where (tC => tC.Index == idx).First ();
+                  else {
+                     conPLine = mPart.PLines.Where (pC => pC.Index == idx).First (); // in case of new edges to be trimmed
+                     foreach (var tC in tempPLines) newPLines.Add (tC);
+                  }
+                  PLine trimmed;
+                  if (pLOrient is Horizontal)
+                     trimmed = conPLine.StartPoint.Y > pLine.StartPoint.Y ? (PLine)conPLine.Trimmed (0, 0, 0, totalBD)
+                                                                          : (PLine)conPLine.Trimmed (0, totalBD, 0, 0);
+                  else
+                     trimmed = conPLine.StartPoint.X > pLine.StartPoint.X ? (PLine)conPLine.Trimmed (0, 0, totalBD, 0)
+                                                                          : (PLine)conPLine.Trimmed (totalBD, 0, 0, 0);
+                  newPLines.Add (trimmed);
+               }
+               break;
+         }
+      }
+   }
+
+   // To be moved to Utils.cs

```

Comment:
Create a method which returns outer pline (compare with the bound points). This method should take parameter as ELoc enum